### PR TITLE
New jwt v2

### PR DESCRIPTION
**Pull request by saroskovic**

### CODE REVIEW

File: src/main/java/axiomq/com/starwars/services/implementations/CustomUserDetails.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Collections;
+
+@Component
+@NoArgsConstructor
+public class CustomUserDetails implements UserDetails {
+
+    private User user;
+
+    public CustomUserDetails(User user) {
+        super();
```
Comment (stefanbesovic) [inline src/main/java/axiomq/com/starwars/services/implementations/CustomUserDetails.java:20]: not needed

File: src/main/java/axiomq/com/starwars/util/JwtUtil.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public String generateToken(String username) {
+        Map<String, Object> claims = new HashMap<>();
+        return createToken(claims, username);
+    }
+
+    private String createToken(Map<String, Object> claims, String subject) {
+
+        return Jwts.builder().setClaims(claims).setSubject(subject).setIssuedAt(new Date(System.currentTimeMillis()))
```
Comment (stefanbesovic) [inline src/main/java/axiomq/com/starwars/util/JwtUtil.java:46]: maybe just to refactor this like: 

```
 .......... .setClaims(claims)
            .setSubject(subject)
            .... etc 
```
In order to be more readable.

File: src/main/java/axiomq/com/starwars/controllers/UserController.java [hunk 3]
```diff
@@ -17,11 +18,13 @@ public class UserController {
 
     private UserService userService;
 
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (stefanbesovic) [inline src/main/java/axiomq/com/starwars/controllers/UserController.java:21]: I think that Branimir told us that we should keep authorization in one place. For example in SecurityConfig simply create new antMatcher for this endpoint.

File: src/main/java/axiomq/com/starwars/config/JwtFilter.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        String authorizationHeader = request.getHeader("Authorization");
+        String token = null;
+        String username = null;
+        if(authorizationHeader != null && authorizationHeader.startsWith("Bearer ")){
+            token = authorizationHeader.substring(7);
+            username = jwtUtil.extractUsername(token);
+        }
+        if(username != null && SecurityContextHolder.getContext().getAuthentication() == null){
```
Comment (branimir-vujicic) [inline src/main/java/axiomq/com/starwars/config/JwtFilter.java:36]: unnecessary block `SecurityContextHolder.getContext().getAuthentication() == null` is always true

File: src/main/java/axiomq/com/starwars/controllers/LoginController.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class LoginController {
+
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
```
Comment (branimir-vujicic) [inline src/main/java/axiomq/com/starwars/controllers/LoginController.java:20]: not used, use custom

**Pull request closed without merging**